File: demo-application/src/main/java/de/jds/Application.java
package de.jds;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.sync.diffsync.PersistenceCallbackRegistry;
import org.springframework.sync.diffsync.config.EnableDifferentialSynchronization;

@EnableDifferentialSynchronization
@SpringBootApplication
public class Application {

  public static void main(String[] args) {
    final ConfigurableApplicationContext run = SpringApplication.run(Application.class, args);
    final PersistenceCallbackRegistry persistenceCallbackRegistry = (PersistenceCallbackRegistry) run.getBean(
        "persistenceCallbackRegistry");
    System.out.println(persistenceCallbackRegistry.findPersistenceCallback("strings"));
  }
}

File: demo-application/src/main/kotlin/de/jds/example/Application.kt
package de.jds.example

import de.sync.core.persistence.PersistenceCallbackRegistry
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.sync.diffsync.config.EnableDifferentialSynchronization

@EnableDifferentialSynchronization
@SpringBootApplication
class Application {
    fun main(args: Array<String>) {
        val run = SpringApplication.run(Application::class.java, *args)
        val persistenceCallbackRegistry = run.getBean(
            "persistenceCallbackRegistry"
        ) as PersistenceCallbackRegistry
        println(persistenceCallbackRegistry.findPersistenceCallback("strings"))
    }
}